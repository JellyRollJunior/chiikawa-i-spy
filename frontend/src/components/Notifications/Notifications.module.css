.notificationWrapper {
  position: absolute;
  z-index: 2;
  display: flex;
  flex-direction: column;
  align-items: end;
  width: 100%;
  gap: 6px;
}
