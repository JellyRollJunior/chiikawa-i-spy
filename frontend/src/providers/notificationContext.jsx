import { createContext } from 'react';

const NotificationContext = createContext({
  notifications: [],
  addPersistentNotification: () => {},
  addTimedNotification: () => {},
  removeNotification: () => {},
});

export { NotificationContext };
