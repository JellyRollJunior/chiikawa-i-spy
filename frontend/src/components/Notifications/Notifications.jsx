import { useContext } from 'react';
import { NotificationContext } from '../../providers/notificationContext.jsx';
import { Notification } from './Notification/Notification.jsx';
import styles from './Notifications.module.css';

const Notifications = ({
  style = { top: '8px', right: '6px', alignItems: 'end' },
}) => {
  const { notifications } = useContext(NotificationContext);

  return (
    <div className={styles.notificationWrapper} style={style}>
      {Array.isArray(notifications) &&
        notifications.length > 0 &&
        notifications.map((notification) => (
          <Notification
            key={notification.id}
            id={notification.id}
            message={notification.message}
            isError={notification.isError}
            isTimed={notification.isTimed}
          />
        ))}
    </div>
  );
};

export { Notifications };
