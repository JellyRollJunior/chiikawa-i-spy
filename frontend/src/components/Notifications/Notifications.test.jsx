import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { Notifications } from './Notifications.jsx';
import { NotificationContext } from '../../providers/notificationContext';

describe('Notifications component', () => {
  it('renders notification wrapper and notification', () => {
    const message = 'hello!';
    const { container } = render(
      <NotificationContext.Provider
        value={{
          notifications: [
            {
              id: 1,
              message,
              isError: false,
              isTimed: false,
            },
          ],
        }}
      >
        <Notifications />
      </NotificationContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
