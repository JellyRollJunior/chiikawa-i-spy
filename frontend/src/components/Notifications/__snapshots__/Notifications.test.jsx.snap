// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Notifications component > renders notification wrapper and notification 1`] = `
<div>
  <div
    class="_notificationWrapper_22a3b0"
    style="top: 8px; right: 6px; align-items: end;"
  >
    <h3
      class="_card_5890fc _timedNotification_5a012d 
      false 
      false"
    >
      <button
        class="_closeButton_5a012d"
      >
        hello!
        <span
          class="_closeButton_5a012d"
        >
          Ã—
        </span>
      </button>
    </h3>
  </div>
</div>
`;
